### PR TITLE
remove redundant CI benchmark check, cleanups

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

re https://github.com/apache/arrow-rs/issues/2149

# Rationale for this change
 
Clean up CI so we don't have to deal with jobs that are not adding value

# What changes are included in this PR?
1. Remove redundeant benchmark check job (it is already covered by other jobs -- I will comment inline)
2. Remove unecessary environment variables from win/mac jobs
3. Removes redundant compile checks in parquet crate

# Are there any user-facing changes?

No

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -44,37 +44,10 @@ jobs:
       - name: Run tests
         shell: bash
         run: |
-          export ARROW_TEST_DATA=$(pwd)/testing/data
-          export PARQUET_TEST_DATA=$(pwd)/parquet-testing/data
           # do not produce debug symbols to keep memory usage down
           export RUSTFLAGS="-C debuginfo=0"
           cargo test
 
-  check_benches:
```
Comment (alamb) [inline .github/workflows/rust.yml:53]: Here is some evidence that this is already covered

When I broke an arrow benchmark:

```shell
echo "fff" >> arrow/benches/string_dictionary_builder.rs
```

Then I ran the check that is part of arrow: https://github.com/apache/arrow-rs/blob/bc493d92c2a032a95d92dab81642f05182f20d81/.github/workflows/arrow.yml#L88

```shell
cargo check -p arrow --all-targets
    Checking arrow v19.0.0 (/Users/alamb/Software/arrow-rs2/arrow)
error: expected one of `!` or `::`, found `<eof>`
  --> arrow/benches/string_dictionary_builder.rs:71:1
   |
71 | fff
   | ^^^ expected one of `!` or `::`

error: could not compile `arrow` due to previous error
warning: build failed, waiting for other jobs to finish...
```

Similarly, when I broke parquet

```shell
echo "ggg" >> parquet/benches/arrow_reader.rs 
```

A check run by the parquet tests also finds it

```shell
cargo check -p parquet --all-features --all-targets
   Compiling parquet v19.0.0 (/Users/alamb/Software/arrow-rs2/parquet)
error: expected one of `!` or `::`, found `<eof>`
   --> parquet/benches/arrow_reader.rs:695:1
    |
695 | ggg
    | ^^^ expected one of `!` or `::`
```

File: .github/workflows/parquet.yml [hunk 2]
```diff
@@ -72,26 +72,17 @@ jobs:
         with:
           rust-version: stable
       - name: Check compilation
-        run: |
```
Comment (alamb) [inline .github/workflows/parquet.yml:75]: I didn't see any reason to run checks that did not have `--all-targets` -- so this change ensures that we test with the following combinations:
1.  default features,
2. `--all-features`
3. `--no-default-features --features=arrow` 

With all targets

Previously, `--all-features --all-targets` was not run checked (and thus the becnhmarks were not checked)

Comment (tustvold) [reply .github/workflows/parquet.yml:75]: This breaks the validation that libraries compile on their own - see #1822

**Review by tustvold — CHANGES_REQUESTED** [review]

File: .github/workflows/parquet.yml (hunk 2 above)
Comment (alamb) [reply .github/workflows/parquet.yml:75]: Got it -- will fix

File: .github/workflows/parquet.yml [hunk 3]
```diff
@@ -76,6 +76,17 @@ jobs:
         uses: ./.github/actions/setup-builder
         with:
           rust-version: stable
+
+        # Run different tests for the library on its own as well as
```
Comment (alamb) [inline .github/workflows/parquet.yml:80]: I tried to clarify the intent of the tests

This also makes it clear (to me) that there should be 8 combinations, and so I also added the missing combination (`--all-targets --all-features`)

File: .github/workflows/parquet.yml (hunk 2 above)
Comment (alamb) [reply .github/workflows/parquet.yml:80]: Fixed in https://github.com/apache/arrow-rs/pull/2212/commits/05986065ce5cdf6a76de646157cc2198f9515028

File: .github/workflows/parquet.yml [hunk 4]
```diff
@@ -91,12 +102,15 @@ jobs:
       - name: Check compilation --all-targets
         run: |
           cargo check -p parquet --all-targets
-      - name: Check compilation --no-default-features --all-targets
+      - name: Check compilation --all-targets --no-default-features
+        run: |
+          cargo check -p parquet --all-targets --no-default-features
+      - name: Check compilation  --all-targets --no-default-features --features-arrow
```
Comment (tustvold) [inline .github/workflows/parquet.yml:108]:
```suggestion
      - name: Check compilation  --all-targets --no-default-features --features arrow
```

**Review by tustvold — APPROVED** [review]

**Pull request merged**